uuid/uuid.go

```diff
@@ -0,0 +1,85 @@
+// Package uuid provides simple UUID generation. Only version 4 style UUIDs
+// can be generated.
+//
+// Please see http://tools.ietf.org/html/rfc4122 for details on UUIDs.
+package uuid
+
+import (
+	"crypto/rand"
+	"io"
+
+	"github.com/Sirupsen/logrus"
+
+	"fmt"
+)
+
+const (
+	// Bits is the number of bits in a UUID
+	Bits = 128
+
+	// Size is the number of bytes in a UUID
+	Size = Bits / 8
+
+	format = "%08x-%04x-%04x-%04x-%012x"
+)
+
+var (
+	// ErrUUIDInvalid indicates a parsed string is not a valid uuid.
+	ErrUUIDInvalid = fmt.Errorf("invalid uuid")
+)
+
+// UUID represents a UUID value. UUIDs can be compared and set to other values
+// and accessed by byte.
+type UUID [Size]byte
+
+// Generate creates a new, version 4 uuid.
+func Generate() (u UUID) {
+	for {
+		// This should never block but the read may fail. Because of this,
+		// we just try to read the random number generator until we get
+		// something. This is a very rare condition but may happen.
+
+		n, err := io.ReadFull(rand.Reader, u[:])
+		if err != nil {
+			logrus.Warnf("error generating version 4 uuid, retrying: %v", err)
```

Comment:
Feels weird to have a logrus dependency in a uuid package. How about having a global variable that defaults to `func(msg string) { fmt.Fprintln(os.Stderr, msg) }` ? clients can then change it in their init() functions.